File: development/backend/const.ts
import path from "path";

// backend level
export const BACKEND_DIR = __dirname
export const DEVELOPMENT_DIR = path.dirname(BACKEND_DIR)

// project level
export const PROJECT_DIR = path.dirname(DEVELOPMENT_DIR)
export const README_PATH = path.join(PROJECT_DIR, "README.md")
export const IMAGES_DIR = path.join(PROJECT_DIR, "images")
export const DATA_DIR = path.join(PROJECT_DIR, "data")
export const DATA_GENERATED_DIR = path.join(DATA_DIR, "generated")
export const DATA_SOURCE_DIR = path.join(DATA_DIR, "source")

// frontend level
export const FRONTEND_DIR = path.join(DEVELOPMENT_DIR, "frontend")
export const FRONTEND_SRC_DIR = path.join(FRONTEND_DIR, "src")
export const FRONTEND_SRC_DATA_DIR = path.join(FRONTEND_SRC_DIR, "data")

File: development/backend/py/src/const.py
import os.path as path


# path

SRC_DIR = path.dirname(__file__)
PROJECT_DIR = path.dirname(SRC_DIR)
ROOT_DIR = path.dirname(path.dirname(path.dirname(PROJECT_DIR)))

DATA_DIR = path.join(ROOT_DIR, "data")
DATA_GEN_DIR = path.join(DATA_DIR, "generated")
DATA_GEN_DETAILED_DIR = path.join(DATA_GEN_DIR, "detailed")
